packages/ecs-agent/ecs.config

```diff
@@ -22,3 +22,9 @@ ECS_CONTAINER_STOP_TIMEOUT="{{settings.ecs.container-stop-timeout}}"
 {{#if settings.ecs.task-cleanup-wait}}
 ECS_ENGINE_TASK_CLEANUP_WAIT_DURATION="{{settings.ecs.task-cleanup-wait}}"
 {{/if}}
+{{#if settings.ecs.image-cleanup-wait}}
+ECS_IMAGE_CLEANUP_INTERVAL="{{settings.ecs.image-cleanup-wait}}"
+{{/if}}
+{{# if settings.ecs.image-cleanup-age}}
+ECS_IMAGE_MINIMUM_CLEANUP_AGE="{{settings.ecs.image-cleanup-age}}"
+{{/if}}
```

Comment:
Don't we also need `ECS_NUM_IMAGES_DELETE_PER_CYCLE` and `ECS_DISABLE_IMAGE_CLEANUP`?

Reply:
They are rendered in the other file. We use `ecs.config` to render configurations that are of type `Duration`. We can't render them in `ecs.config.json` because the parsing library used by the ECS agent doesn't call the right method while deserializing strings in the format of `1h`, `1m`, etc.

Reply:
OK, thanks for the explanation. I don't see where those particular values are rendered, but I trust things are as expected.

Reply:
This is where the new settings are added to the struct that `ecs-settings-applier` renders (it references a commit outside the Bottlerocket repo, so the link might be broken in the future):

https://github.com/bottlerocket-os/bottlerocket/blob/0a5a17e515325c9505ef6bf6d8be7b09f39ac970/sources/api/ecs-settings-applier/src/ecs.rs#L153

And this is where the struct is rendered to disk:

https://github.com/bottlerocket-os/bottlerocket/blob/a4a10399177245be08bde00aa9c6af843863bacb/sources/api/ecs-settings-applier/src/ecs.rs#L161

Reply:
Yeah, I saw that part, but didn't see how the serialization translates `image_cleanup_disabled` to `ECS_DISABLE_IMAGE_CLEANUP` and `NumImagesToDeletePerCycle` to `ECS_NUM_IMAGES_DELETE_PER_CYCLE`. But I see there's a difference between the `ecs.config`and `ecs.config.json` files. 👍